### PR TITLE
Resolve #18 Add gerrit label support

### LINKED ISSUE #18: Add gerrit label support

### PR DESCRIPTION
**Pull request by rolyatwilson**

### CODE REVIEW

File: lib/xlint.rb [hunk 1]
```diff
@@ -28,6 +29,8 @@ def check_env
 
     def build_draft
       @comments = []
+      # GitDiffParser::Patches.parse(cp932 text) raises ArgumentError: invalid byte sequence in UTF-8
```
Comment (bootstraponline) [inline lib/xlint.rb]: When does this error happen? Is there another diff parser we could use that's more robust?

File: lib/xlint.rb [hunk 2]
```diff
@@ -101,7 +111,7 @@ def check_deployment_target(change)
           path: change[:file],
           position: change[:line_number],
           message: 'Deployment target changes should be approved by the team lead.',
-          severity: 'error'
+          severity: 'warn'
```
Comment (bootstraponline) [inline lib/xlint.rb]: I think we still want severity error? just with a `-1` gerrit value.

File: lib/xlint.rb (hunk 1 above)
Comment (bootstraponline) [inline lib/xlint.rb]: [canvas-lms](https://github.com/instructure/canvas-lms/blob/58912ddfb899a1b1c151c3c64e872b13300468c8/gems/rubocop-canvas/lib/rubocop_canvas/helpers/diff_parser.rb) appears to have their own parser.

Comment (bootstraponline) [inline lib/xlint.rb]: http://stackoverflow.com/a/18454435

Comment (rolyatwilson) [inline lib/xlint.rb]: I ran into the error when the git diff contained a PDF file. It's probable image files would also trigger the error.

File: lib/xlint.rb (hunk 2 above)
Comment (rolyatwilson) [inline lib/xlint.rb]: Gergich will use the lowest value in the draft as the score it posts. On the Jenkins job I'm testing with, having this set to error prevented Gergich from posting to the Lint-Review column.

Comment (bootstraponline) [inline lib/xlint.rb]: With the new gerrit update, error should work again.

File: lib/xlint.rb [hunk 3]
```diff
@@ -28,7 +29,11 @@ def check_env
 
     def build_draft
       @comments = []
-      diff = Xlint.parse_git(File.read(diff_file))
+      # GitDiffParser::Patches.parse(cp932 text) raises ArgumentError: invalid byte sequence in UTF-8
+      # https://github.com/packsaddle/ruby-git_diff_parser/issues/91
+      diff_data = File.read(diff_file)
+      diff_data = diff_data.encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '')
```
Comment (bootstraponline) [inline lib/xlint.rb]: `diff_data.encode!` is cleaner since we can drop the `diff_data =`

**Pull request merged**